### PR TITLE
fix: less noisy CSS unused rules

### PR DESCRIPTION
**Pull request by patrickhulce**
* filter out duplicate stylesheets
* don't report stylesheets that have no rules

addresses #1456

### CODE REVIEW

**Comment by patrickhulce** [comment]
R: @ebidel 

FWIW, I still think flagging the sheets as duplicate is cleaner than not returning them at all, but I totally empathize with the unnecessary noisiness :)

File: lighthouse-cli/test/fixtures/dobetterweb/dbw_tester.html [hunk 1]
```diff
@@ -101,6 +101,15 @@
     font-size: 20px;
   }
   </style>
+
+  <style>
+  /* Empty on purpose */
```
Comment (ebidel) [inline lighthouse-cli/test/fixtures/dobetterweb/dbw_tester.html]: /* PASS - Empty on purpose */

File: lighthouse-cli/test/fixtures/dobetterweb/dbw_tester.html [hunk 2]
```diff
@@ -101,6 +101,15 @@
...
+  </style>
+
+  <style>
+  /* Just an import shouldn't show up */
```
Comment (ebidel) [inline lighthouse-cli/test/fixtures/dobetterweb/dbw_tester.html]: PASS - Just an import...

File: lighthouse-core/audits/unused-css-rules.js [hunk 3]
```diff
@@ -127,7 +132,7 @@ class UnusedCSSRules extends Audit {
     const unusedRatio = (unused / usage.length) || 0;
     const results = Object.keys(indexedSheets).map(sheetId => {
       return UnusedCSSRules.mapSheetToResult(indexedSheets[sheetId]);
-    });
+    }).filter(Boolean);
```
Comment (ebidel) [inline lighthouse-core/audits/unused-css-rules.js]: Don't feel strongly about it, but `.filter(n => n)` would also filter out the falsey values.

**Comment by ebidel** [comment]
As an example, here's a run from chromestatus.com. It has a bunch of duplicate sheets from shadow dom usage:

screen shot 2017-01-13 at 10 35 39 am

File: lighthouse-core/audits/unused-css-rules.js (hunk 3 above)
Comment (patrickhulce) [inline lighthouse-core/audits/unused-css-rules.js]: Yeah I guess this is a pattern I'm used to, but is there a reason we don't use `Boolean` anywhere? `!!` seems to be preferred for casting throughout.

File: lighthouse-cli/test/fixtures/dobetterweb/dbw_tester.html (hunk 1 above)
Comment (patrickhulce) [inline lighthouse-cli/test/fixtures/dobetterweb/dbw_tester.html]: done

File: lighthouse-cli/test/fixtures/dobetterweb/dbw_tester.html (hunk 2 above)
Comment (patrickhulce) [inline lighthouse-cli/test/fixtures/dobetterweb/dbw_tester.html]: done

**Comment by patrickhulce** [comment]
PTAL :)

**Pull request merged**